src/main/java/de/dhbw/mh/slang/codegen/JavaBytecodeGenerator.java

```diff
@@ -59,12 +59,25 @@ public String visitPost( AstUnaryOperation node, String base ){
 
 	@Override
 	public String visitPost( AstBinaryOperation node, String lhs, String rhs ){
+		String prefix= "";
+		switch (node.getDatatype()) {
+			case BOOL: prefix = "i";break;
+			case I8: prefix = "i";break;
+			case I16: prefix = "i";break;
+			case I32: prefix = "i";break;
+			case I64: prefix = "l";break;
+			case F32: prefix = "f";break;
+			case F64: prefix = "d";break;		
+			default:
+				prefix = "i";
+				break;
+		}
 		switch( node.OPERATOR ){
 			case ADD:{
 				throw new RuntimeException( "not yet implemented" );
 			}
 			case SUBTRACT:{
-				throw new RuntimeException( "not yet implemented" );
+				return String.format("LHS%nRHS%n" + prefix + "sub%n");
```

Comment:
LHS and RHS represent the instructions of the left and right nodes. These are passed to the method via the parameters `lhs` and `rhs`. My test infrastructure eliminates these dependencies, so the strings LHS and RHS are passed generically. Therefore, please use the parameters instead of a literal so that the integration of all team results works.